src/system.rs

```diff
@@ -1640,6 +1648,24 @@ mod schema_system_tests {
         assert!(schema.is_ok());
     }
 
+    #[test]
+    fn top_level_type_def_without_name_field() {
```

Comment:
question—what happens if more than one `name` field is present, or if the `name` field is not a symbol?

Reply:
If more than one `name` field are supplied it would select the last one. If the `name` field is not a `string` or `symbol` it will return an error.

Reply:
>  If more than one name field are supplied it would select the last one.

Structs are unordered, so this is undefined behavior. It needs to be an error if more than one name field exists.

> If the name field is not a string or symbol it will return an error.

I'm pretty sure the spec says that type names may only symbols, so it should also be an error if the type name is a string.

Reply:
Reference PR: #125